packages/syft/src/syft/service/user/user_service.py

```
@@ -175,6 +175,20 @@ def search(
 
         return results
 
+    @service_method(
+        path="user.get_current_user", name="get_current_user", roles=GUEST_ROLE_LEVEL
+    )
+    def get_current_user(self, context: AuthedServiceContext) -> UserView:
+        result = self.stash.get_by_verify_key(
+            credentials=context.credentials, verify_key=context.credentials
+        )
+        if result.is_ok():
+            # this seems weird that we get back None as Ok(None)
```

It returns None if no user if found with given verify key

---

so should we return an error?

---

Yes, we can return an error at the user service level,